src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -187,7 +187,7 @@ private IAsyncCollector<IStorageQueueMessage> BuildFromQueueAttribute(QueueAttri
 
             internal IStorageQueue GetQueue(QueueAttribute attrResolved)
             {
-                var account = Task.Run(() => this._accountProvider.GetStorageAccountAsync(attrResolved, CancellationToken.None)).GetAwaiter().GetResult();
```

Comment:
This is the correct way to run await an async task from an sync method and avoid deadlocks. 

---
In reply to: [122791012](https://github.com/Azure/azure-webjobs-sdk/pull/1190#discussion_r122791012) [](ancestors = 122791012)